Last version of paho-mqtt dependency causes crash on startup

# Problem/Motivation

> mqtt-io  uses very old version of asyncio-mqtt (8.1), depending on `paho-mqtt>=1.5.0`. In Feb 2024 paho-mqtt released version 2.0 introducing some breaking changes, resulting in following error during startup of the addon:

```
2024-06-04 00:16:17 mqtt_io.__main__ [ERROR] MqttIo crashed!
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/mqtt_io/__main__.py", line 107, in main
    mqtt_gpio.run()
  File "/usr/lib/python3.11/site-packages/mqtt_io/server.py", line 1280, in run
    self.loop.run_until_complete(self._main_task)
  File "/usr/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/mqtt_io/server.py", line 1199, in _main_loop
    await self._connect_mqtt()
  File "/usr/lib/python3.11/site-packages/mqtt_io/server.py", line 596, in _connect_mqtt
    await self.mqtt.connect()
  File "/usr/lib/python3.11/site-packages/mqtt_io/mqtt/asyncio_mqtt.py", line 32, in inner
    await func(*args, **kwargs)
  File "/usr/lib/python3.11/site-packages/mqtt_io/mqtt/asyncio_mqtt.py", line 79, in connect
    await self._client.connect(timeout=timeout)
  File "/usr/lib/python3.11/site-packages/asyncio_mqtt/client.py", line 82, in connect
    if not isinstance(client_socket, mqtt.WebsocketWrapper):
```

## Steps to reproduce

> Reinstall (rebuild) addon on latest version of HA OS / HA CORE

## Proposed changes

> The issue should be fixed [upstream](https://github.com/flyte/mqtt-io/issues/348). But it looks that following line added to addon's `requirements.txt` solves the problem for now:
```
paho-mqtt<2
```

--- a/mqtt-io/requirements.txt
+++ b/mqtt-io/requirements.txt
@@ -6,6 +6,7 @@ bme680==1.1.1
 gpiod==2.1.3
 gpiozero==2.0.1
 mqtt-io @ git+https://github.com/flyte/mqtt-io@2bb2f945ea47ed87f99ab4fc966d2dbbd47fc7c7
+paho-mqtt<2
 nfcpy==1.0.4
 pcf8574==0.1.3
 pcf8575==0.3

